include/rdma/fi_var.h

```diff
@@ -40,6 +40,13 @@
 extern "C" {
 #endif
 
+struct fi_env {
+	const struct fi_provider *provider;
```

Comment:
I included the fi_provider pointer so we could pass it to the various fi_var_get_* functions.